src/gui/qgsmaptoolcapture.cpp

```diff
@@ -483,7 +483,34 @@ int QgsMapToolCapture::fetchLayerPoint( const QgsPointLocator::Match &match, Qgs
           return 2;
         QgsLineString line( geom.constGet()->vertexAt( vId ), geom.constGet()->vertexAt( vId2 ) );
 
-        layerPoint = QgsGeometryUtils::closestPoint( line,  QgsPoint( match.point() ) );
+        QgsPoint pt( match.point() );
+        // Transform point to sourceLayer crs, since vId and vId2 coordinates are in sourceLayer crs
+        if ( sourceLayer->crs() != QgsProject::instance()->crs() )
+        {
+          try
+          {
+            pt.transform( QgsCoordinateTransform( QgsProject::instance()->crs(), sourceLayer->crs(), sourceLayer->transformContext() ) );
```

Comment:
you could save a bit of work here by making the QgsCoordinateTransform once and storing it in a variable, and then using the reverse transform flag when reprojecting back. Creation of QgsCoordinateTransform isn't cheap.

Reply:
I can't because it's not the same transformation.

Reply:
Oops, yep